### PR TITLE
add by attribute sorting capability

### PR DESCRIPTION
**Pull request by quazardous**
replace #2 because revised master was too different

### CODE REVIEW

**Comment by quazardous** [comment]
I've not updated the README.md...

File: lib/readify.js [hunk 1]
```diff
@@ -19,8 +19,43 @@ const good = (f) => (...a) => f(null, ...a);
 
 const sort = currify((fn, array) => array.sort(fn));
 /* sorting on Win and node v0.8.0 */
-const sortFiles = sort((a, b) => {
-    return a.name > b.name ? 1 : -1;
+const sortFiles = currify((attr, order, array) => {
+    const cmpCallbacks = {
+        'numeric': (a, b) => (+a - +b),
+        'default': (a, b) => (a > b ? 1 : -1),
```
Comment (coderaiser) [inline lib/readify.js:25]: What difference between `numeric` and `default`?

File: lib/readify.js [hunk 2]
```diff
@@ -19,8 +19,43 @@ const good = (f) => (...a) => f(null, ...a);
...
+    switch (order) {
+    case 'asc':
+        // nothing
+        break;
+    case 'desc':
+        // nothing
+        break;
+    default:
+        order = 'asc';
+    }
```
Comment (coderaiser) [inline lib/readify.js]: Maybe is better to use `if` instead of  `switch` in this condition?

**Comment by coderaiser** [comment]
Thank you :), merged as e51fa08.

**Comment by coderaiser** [comment]
Landed in [v4.1.0](https://github.com/coderaiser/readify/releases/tag/v4.1.0).

**Comment by coderaiser** [comment]
Added ability to sort directories by date in [v4.1.5](https://github.com/coderaiser/readify/releases/tag/v4.1.5).

**Pull request closed without merging**